### PR TITLE
color and boxplot updates

### PR DESCRIPTION
**Pull request by marisand**
olink_color: New color scheme (gray to lightblue).
olink_boxplot: Can now take a character vector as grouping variable. First value used as x-axis, second as fill (legend). 
Bugfix for verbose=T to plot all plots, not just last one.

### CODE REVIEW

**Comment by kathy-nevola** [comment]
Do we need a unit test for this? Is this something we want/need to implement immediately? If not, we should merge into develop instead and the branch should come off of develop. Additionally, dont forget to import selectively if any changes have been made to the functions that are called?

File: OlinkAnalyze/R/Olink_color.R [hunk 1]
```diff
@@ -2,14 +2,14 @@
 #'
 #' @param alpha transparency (optional)
 #' @param coloroption string, one or more of the following:
-#' c('red', 'orange', 'yellow', 'green', 'teal', 'turqoise', 'gray', 'darkblue', 'purple', 'pink')
```
Comment (kathy-nevola) [inline OlinkAnalyze/R/Olink_color.R:5]: Do we need to change any reference files for the unit tests for this?  I dont think we have a unit test for colors so probably not.

File: OlinkAnalyze/R/Olink_boxplot.R [hunk 2]
```diff
@@ -73,9 +74,15 @@ olink_boxplot <- function(df,
...
 
   }
-
-
+  
   #Setup
+  x_variable <- rlang::syms(variable[1])
+  if(length(variable) > 1){
+    fill_variable <- rlang::syms(variable[2])
+  }else{
```
[the remark this answers is not in the document]
Comment (marisand) [reply OlinkAnalyze/R/Olink_boxplot.R:88]: Fixed

File: OlinkAnalyze/R/Olink_color.R (hunk 1 above)
Comment (marisand) [reply OlinkAnalyze/R/Olink_color.R:5]: We probably should add unit tests for color.

File: OlinkAnalyze/R/Olink_boxplot.R [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
+#'Generates faceted boxplots of NPX vs. grouping variable(s) for a given list of proteins (OlinkIDs) using ggplot and ggplot2::geom_boxplot.
 #'
-#' @param df NPX data frame in long format with at least protein name (Assay), OlinkID (unique), UniProt and a grouping variable.
-#' @param variable  A character value indiciating which column to use as the x-axis grouping variable
+#' @param df NPX data frame in long format with at least protein name (Assay), OlinkID (unique), UniProt and at least one grouping variable.
+#' @param variable  A character vector or character value indicating which column to use as the x-axis and fill grouping variable. 
+#' The first or single value is used as x-axis, the second as fill. Further values in a vector are not plotted.
 #' @param olinkid_list Character vector indicating which proteins (OlinkIDs) to plot.
 #' @param number_of_proteins_per_plot Number of boxplots to include in the facet plot (default 6).
 #' @param verbose Boolean. If the plots are shown as well as returned in the list (default is false).
```
[the remark this answers is not in the document]
Comment (marisand) [reply OlinkAnalyze/R/Olink_boxplot.R:10]: Check!

File: OlinkAnalyze/R/Olink_color.R (hunk 1 above)
Comment (kathy-nevola) [reply OlinkAnalyze/R/Olink_color.R:5]: I will discuss with @Orbmac  for implementation in a future update and add to Kanban

File: OlinkAnalyze/R/Olink_boxplot.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (kathy-nevola) [reply OlinkAnalyze/R/Olink_boxplot.R:10]: Will fix this in merging

**Review by kathy-nevola — APPROVED** [review]
Looks good! Thanks for all your changes. Will discuss with @Orbmac and team about new unit tests for future update

**Pull request merged**